### PR TITLE
fix: extract timeout from context

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

File: wakuv2/nwaku.go [hunk 1]
```diff
@@ -3142,3 +3146,11 @@ func (n *WakuNode) GetNumConnectedPeers() (int, error) {
 	}
 	return len(peers), nil
 }
+
+func getContextTimeoutMilliseconds(ctx context.Context) int {
+	deadline, ok := ctx.Deadline()
+	if ok {
+		return int(time.Until(deadline).Milliseconds())
+	}
+	return 0
```
Comment (Ivansete-status) [inline wakuv2/nwaku.go:3129]: I wonder if we should make the app to crash earlier. Maybe adding a log before this line and/or returning -1 may help

**Review by Ivansete-status — APPROVED** [review]
LGTM! Thanks for it! :100: 
I just added a nitpick comment ;P

**Review by gabrielmer — APPROVED** [review]
LGTM!

File: wakuv2/nwaku.go (hunk 1 above)
Comment (richard-ramos) [reply wakuv2/nwaku.go:3129]: ah! the 0 is not an invalid error! it means that it's a context without deadline, and a timeoutMs of 0 is equal to not having timeout!

**Pull request merged**